Mathlib/LinearAlgebra/Matrix/NonsingularInverse.lean

```diff
@@ -607,6 +611,24 @@ theorem inv_diagonal (v : n → α) : (diagonal v)⁻¹ = diagonal (Ring.inverse
 
 end Diagonal
 
+/-- The inverse of a 1×1 or 0×0 matrix is always diagonal.
+
+While we could write this as `of fun _ _ => Ring.inverse (A default default)` on the RHS, this is
+less useful because:
+
+* It wouldn't work for 0×0 matrices.
+* More things are true about diagonal matrices than constant matrices, and so more lemmas exist.
+
+`Matrix.diagonal_unique` can be used to reach this form, while `Ring.inverse_eq_inv` can be used
+to replace `Ring.inverse` with `⁻¹`.
+-/
+@[simp]
+theorem inv_subsingleton [Subsingleton m] [Fintype m] [DecidableEq m] (A : Matrix m m α) :
+    A⁻¹ = diagonal fun i => Ring.inverse (A i i) := by
```

Comment:
I'm not sure this should be `simp` - it'll fire any time there's an inverse matrix around, and the search for `Subsingleton m` could be expensive

Reply:
`Matrix.det_unique` is already simp; are the considerations for `Unique` different to those for `subsingleton?

Reply:
Potentially, but I'd guess the larger problem is that `det` shows up a little less often than `⁻¹` on matrices